server/functions/src/index.ts

```diff
@@ -0,0 +1,126 @@
+import * as admin from "firebase-admin";
+import * as functions from "firebase-functions";
+import * as fs from "fs";
+
+admin.initializeApp();
+const db = admin.firestore();
+
+const COUNTRY_CODE = /^[A-Z][A-Z]$/; // Literal regex for immediate compilation.
+
+// FCM documentation doesn"t specify standard but usually up to 162 chars
+// https://stackoverflow.com/a/12502351/1509221
+const FCM_TOKEN_MAX_LENGTH = 4096;
+
+enum Platform {
+  IOS = 0,
+  ANDROID = 1,
+  WEB = 2,
+}
+
+interface Client {
+  uuid: string;
+  token: string;
+  disableNotifcations: boolean;
+  platform: Platform;
+  isoCountryCode: string;
+  subscribedTopics: string[];
+}
+
+// Implementation of the v1 API"s `getCaseStats` method.
+// TODO: replace with direct Firestore access from the client.
+export const getCaseStats = functions
+  .region("europe-west6")
```

Comment:
configuration question: what does this do? the code is specifying a datacenter which it runs in somehow?  what happens if we want to run ain instance in us-north-27 or whatever? (note: unlikley to happen but... still seems odd)

Reply:
Yeah, this indeed is "infrastructure configuration as code" - it's analogous to writing a Terraform config telling us where to deploy the code, but does it inline in the source code itself.

If we always want to deploy to multiple regions, we can also pass a list of regions.

If we want to deploy some project IDs to region X and others to region Y, we can compute the value given to "region" inline before exporting the function.

If we hate the inline configuration inside source code, we can drop the Firebase tooling and deploy the Cloud Functions through standard `gcloud` tooling and Terraform.